assets/app/view/spreadsheet.rb

```diff
@@ -96,14 +103,71 @@ def render_title
       ]
     end
 
+    def render_sort_link(props, title, sort_by)
+      h(:th, props, [
+        h(
+          Link,
+          href: '',
+          click: lambda {
+            mark_sort_column(sort_by)
+            toggle_sort_order
+          },
+          children: title + (@spreadsheet_sort_by == sort_by ? ' ' + sort_order_icon : ''),
+          class: ''
+        ),
+      ])
+    end
+
+    def sort_order_icon
+      return '(⬇️)' if @spreadsheet_sort_order == 'ASC'
+
+      '(⬆️)'
+    end
+
+    def mark_sort_column(sort_by)
+      Lib::Storage['spreadsheet_sort_by'] = sort_by
+      update
+    end
+
+    def toggle_sort_order
+      Lib::Storage['spreadsheet_sort_order'] = 'ASC' if @spreadsheet_sort_order == 'DESC'
+      Lib::Storage['spreadsheet_sort_order'] = 'DESC' unless @spreadsheet_sort_order == 'DESC'
+      update
+    end
+
     def render_corporations
       current_round = @game.round.turn_round_num
 
+      ordered_corporations = sorted_corporations
+      ordered_corporations.map do |c|
+        render_corporation(c[1], c[0], current_round)
+      end
+    end
+
+    def sorted_corporations
+      result = []
       floated_corporations = @game.round.entities
+
       @game.corporations.map do |c|
```

Comment:
if you're doing map, there's no reason to have an array

result = @game.corporations.map do |c|
[operating_order, c]
end